packages/nifti-volume-loader/src/helpers/fetchAndAllocateNiftiVolume.ts

```diff
@@ -0,0 +1,263 @@
+import * as NiftiReader from 'nifti-reader-js';
+import {
+  cache,
+  utilities,
+  Enums,
+  eventTarget,
+  triggerEvent,
+} from '@cornerstonejs/core';
+import type { Types } from '@cornerstonejs/core';
+import { vec3 } from 'gl-matrix';
+import makeVolumeMetadata from './makeVolumeMetadata';
+import NiftiImageVolume from '../NiftiImageVolume';
+import * as NIFTICONSTANTS from './niftiConstants';
+import { invertDataPerFrame, rasToLps } from './convert';
+import scaleNiftiArray from './scaleNiftiArray';
+import Events from '../enums/Events';
+
+const { createUint8SharedArray, createFloat32SharedArray } = utilities;
+
+export const urlsMap = new Map();
+
+function fetchArrayBuffer(url, onProgress, signal, onload) {
+  return new Promise((resolve, reject) => {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', url, true);
+
+    xhr.responseType = 'arraybuffer';
+
+    xhr.onload = function (e) {
+      if (onload && typeof onload === 'function') {
+        onload();
+      }
+
+      resolve(xhr.response);
+    };
+
+    if (onProgress && typeof onProgress === 'function') {
+      xhr.onprogress = function (e) {
+        //  if (e.lengthComputable) {
```

Comment:
Consider uncommenting this code as the `lengthComputable` flag indicates if `loaded` and `total` are "useable".